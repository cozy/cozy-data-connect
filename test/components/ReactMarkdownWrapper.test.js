'use strict'

/* eslint-env jest */

import React from 'react'
import { configure, shallow } from 'enzyme'
import Adapter from 'enzyme-adapter-react-16'

import {
  ReactMarkdownWrapper,
  reactMarkdownRendererOptions
} from '../../src/components/ReactMarkdownWrapper'

configure({ adapter: new Adapter() })

describe('ReactMarkdownWrapper component', () => {
  beforeEach(() => {
    jest.resetModules()
  })

  it('should be displayed correctly if source provided', () => {
    const component = shallow(
      <ReactMarkdownWrapper source={'**test** using `markdown`'} />
    ).getElement()
    expect(component).toMatchSnapshot()
  })

  it('should be displayed correctly if source with link', () => {
    const component = shallow(
      <ReactMarkdownWrapper
        source={
          '**test** using [markdown](https://en.wikipedia.org/wiki/Markdown)'
        }
      />
    ).getElement()
    expect(component).toMatchSnapshot()
  })
})

describe('ReactMarkdown options', () => {
  it('should correctly return an anchor HTML element', () => {
    expect(
      reactMarkdownRendererOptions.Link({
        href: 'https://testlink.mock',
        children: 'example link test for test'
      })
    ).toMatchSnapshot()
  })
})
