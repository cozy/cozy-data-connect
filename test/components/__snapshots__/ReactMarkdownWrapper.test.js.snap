// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ReactMarkdown options should correctly return an anchor HTML element 1`] = `
<a
  href="https://testlink.mock"
  target="_blank"
>
  example link test for test
</a>
`;

exports[`ReactMarkdownWrapper component should be displayed correctly if source provided 1`] = `
<ReactMarkdown
  containerTagName="div"
  parserOptions={Object {}}
  renderers={
    Object {
      "Link": [Function],
    }
  }
  source="**test** using \`markdown\`"
/>
`;

exports[`ReactMarkdownWrapper component should be displayed correctly if source with link 1`] = `
<ReactMarkdown
  containerTagName="div"
  parserOptions={Object {}}
  renderers={
    Object {
      "Link": [Function],
    }
  }
  source="**test** using [markdown](https://en.wikipedia.org/wiki/Markdown)"
/>
`;
